test/models/stacks_test.rb

```diff
@@ -729,5 +730,105 @@ def setup
 
       @stack.async_refresh_deployed_revision
     end
+
+    test "#lock_reverted_commits! locks all commits between the original and reverted commits" do
```

Comment:
Ended up adding a bunch of test coverage here as there's some in https://github.com/Shopify/shipit-engine/blob/22403f4c7d51e70e9d3c0c075afac96c7e0d7584/test/jobs/github_sync_job_test.rb#L39-L87 and https://github.com/Shopify/shipit-engine/blob/22403f4c7d51e70e9d3c0c075afac96c7e0d7584/test/models/rollbacks_test.rb#L22-L56 but not enough to convince me that I wasn't breaking anything with the refactor.

I've also split the addition of these tests and the refactor into separate commits so it's more obvious what my changes actually did.